instagram-app/src/components/PostContainer/PostsPage.js

```diff
@@ -0,0 +1,98 @@
+import React, { Component } from "react";
+import Fuse from "fuse.js";
+import data from "../../dummy-data";
+import SeachBar from "../SearchBar/SearchBar";
+import PostContainer from "./PostContainer";
+
+export default class PostsPage extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      posts: [],
+      searchInput: ""
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener("beforeunload", () => {
+      this.updateLocalStorage();
+    });
+
+    let initialPosts;
+
+    try {
+      initialPosts = JSON.parse(localStorage.instagramClone);
+    } catch (e) {
+      initialPosts = data;
+    }
+
+    this.setState({
+      posts: initialPosts
+    });
+  }
+
+  likeComment = id => {
+    const newPosts = [...this.state.posts];
+
+    newPosts.forEach(post => {
+      if (post.id === id) {
+        if (!post.liked) {
+          post.likes += 1;
+          post.liked = !post.liked;
+        } else {
+          post.likes -= 1;
+          post.liked = !post.liked;
+        }
+      }
+    });
+
+    this.setState({
+      posts: newPosts
+    });
+  };
+
+  searchInputChange = e => {
+    this.setState({
+      searchInput: e.target.value
+    });
+  };
+
+  updateLocalStorage = () => {
+    localStorage.instagramClone = JSON.stringify(this.state.posts);
+  };
+
+  render() {
+    const { posts, searchInput } = this.state;
+    const options = {
```

Comment:
Why you are using hooks here, inside your render method.

Reply:
Not hooks, I destructured state.

Reply:
You are  very right, this happens when you do reviews while you are in a zoom meeting.